app/services/password_changer.go

```diff
@@ -24,5 +26,16 @@ func PasswordChanger(store data.AccountStore, r ops.ErrorReporter, cfg *app.Conf
 		return FieldErrors{{"credentials", ErrFailed}}
 	}
 
+	//Check TOTP MFA
+	if account.TOTPEnabled() {
+		secret, err := compat.Decrypt([]byte(account.TOTPSecret.String), cfg.DBEncryptionKey)
+		if err != nil {
+			return errors.Wrap(err, "TOTPDecrypt")
+		}
+		if !totp.Validate(totpCode, secret) {
+			return FieldErrors{{"totp", ErrInvalidOrExpired}}
+		}
+	}
+
```

Comment:
we decided in the previous PR to remove this requirement, since PasswordChanger is called when users already have an authenticated session.

https://github.com/keratin/authn-server/pull/133#discussion_r342647440